conan/api/output.py

```diff
@@ -204,11 +209,16 @@ def success(self, msg):
         return self
 
     def warning(self, msg, warn_tag=None):
-        if self._conan_output_level <= LEVEL_WARNING:
-            if warn_tag is not None and warn_tag in self._silent_warn_tags:
+        def _skip_warning():
+            return (warn_tag is not None and warn_tag in self._silent_warn_tags
+                    or warn_tag is None and "unknown" in self._silent_warn_tags)
+        if self._conan_output_level <= LEVEL_WARNING or (self._warnings_as_errors and self._conan_output_level <= LEVEL_ERROR):
+            if _skip_warning():
                 return self
             warn_tag_msg = "" if warn_tag is None else f"{warn_tag}: "
-            self._write_message(f"WARN: {warn_tag_msg}{msg}", Color.YELLOW)
+            message_type = "ERROR" if self._warnings_as_errors else "WARN"
+            message_color = Color.RED if self._warnings_as_errors else Color.YELLOW
+            self._write_message(f"{message_type}: {warn_tag_msg}{msg}", message_color)
```

Comment:
Not raising?

Reply:
We need to check this in more depth. If we raise here, it begs the question as to why normal `error`s do not raise in some cases

Reply:
``error()`` messages should be there mostly for 2 reasons:
- They will end up raising an exception, but we want to print a bunch of error messages before raising the exception instead of creating a huge exception message, sometimes a bit more difficult to read
- They are legacy, something was bad, but we couldn't really raise because of risk of breaking users, but we still wanted to print a red error message.

So maybe we can assume that module the legacy, error messages are only printed before exceptions?